parsely/src/main/java/com/parsely/parselyandroid/FlushManager.kt

```diff
@@ -0,0 +1,41 @@
+package com.parsely.parselyandroid
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.isActive
+import kotlinx.coroutines.launch
+
+/**
+ * Manager for the event flush timer.
+ *
+ *
+ * Handles stopping and starting the flush timer. The flush timer
+ * controls how often we send events to Parse.ly servers.
+ */
+internal class FlushManager(
+    private val parselyTracker: ParselyTracker,
+    val intervalMillis: Long,
+    private val coroutineScope: CoroutineScope
+) {
+    private var job: Job? = null
+
+    fun start() {
+        if (job?.isActive == true) {
+            return
+        }
+        job = coroutineScope.launch {
```

Comment:
Question (❓): How confident are we that the `Timer` related implementation and the `Coroutines` related one are equivalent, from 1 to 10, like 10? 😊

```kotlin
        runningTask = object : TimerTask() {
            override fun run() {
                parselyTracker.flushEvents()
            }
        }
        parentTimer.scheduleAtFixedRate(runningTask, intervalMillis, intervalMillis)
```

vs

```kotlin
        job = coroutineScope.launch {
            while (isActive) {
                delay(intervalMillis)
                parselyTracker.flushEvents()
            }
        }
```

Reply:
Hm, I'm not sure how I could rate - do you have some concerns? If yes, we can add a test for concerning scenario. I think the code in Coroutines will behave exactly the same, including initial delay (second parameter of the method, [docs](https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/util/Timer.html#scheduleAtFixedRate(java.util.TimerTask,%20long,%20long)))

Reply:
> Hm, I'm not sure how I could rate

From your answer it seems that you are fully confident, like a 10? 😊 

> do you have some concerns? If yes, we can add a test for concerning scenario.

No particular concerns other than me never had to do this `Timer` to `Coroutines` conversion, and me not going to deep into the implementation details of how the `Timer` works other than reading the documentation. 🤷 

> I think the code in Coroutines will behave exactly the same, including initial delay (second parameter of the method, [docs](https://docs.oracle.com/javase%2F7%2Fdocs%2Fapi%2F%2F/java/util/Timer.html#scheduleAtFixedRate(java.util.TimerTask,%20long,%20long)))

Yeap, it does seems so indeed... 🤞 

-----

FYI: I am resolving this, no need for further conversation. 😊